tools/config.c

```diff
@@ -172,14 +187,35 @@ main (gint argc, gchar** argv)
 		return 1;
 	}
 
-	if ((opt_user && opt_system)
-	    || (opt_read && (opt_servers_object != NULL || opt_servers_kv != NULL || opt_object_backend != NULL || opt_object_component != NULL || opt_object_path != NULL || opt_kv_backend != NULL || opt_kv_component != NULL || opt_kv_path != NULL))
-	    || (opt_read && !opt_user && !opt_system)
-	    || (!opt_read && (opt_servers_object == NULL || opt_servers_kv == NULL || opt_object_backend == NULL || opt_object_component == NULL || opt_object_path == NULL || opt_kv_backend == NULL || opt_kv_component == NULL || opt_kv_path == NULL))
-	    || opt_max_operation_size < 0
-	    || opt_max_connections < 0
-	    || opt_stripe_size < 0
-	)
+	if ((opt_user && opt_system) //
+		|| (opt_read && (opt_servers_object != NULL //
+					|| opt_servers_kv != NULL //
+					|| opt_servers_db != NULL //
+					|| opt_object_backend != NULL //
+					|| opt_object_component != NULL //
+					|| opt_object_path != NULL //
+					|| opt_kv_backend != NULL //
+					|| opt_kv_component != NULL //
+					|| opt_kv_path != NULL //
+					|| opt_db_backend != NULL //
+					|| opt_db_component != NULL //
+					|| opt_db_path != NULL)) //
+		|| (opt_read && !opt_user && !opt_system) //
+		|| (!opt_read && (opt_servers_object == NULL //
+					 || opt_servers_kv == NULL //
+					 || opt_servers_db == NULL //
+					 || opt_object_backend == NULL //
+					 || opt_object_component == NULL //
+					 || opt_object_path == NULL //
+					 || opt_kv_backend == NULL //
+					 || opt_kv_component == NULL //
+					 || opt_kv_path == NULL //
+					 || opt_db_backend == NULL //
+					 || opt_db_component == NULL //
+					 || opt_db_path == NULL)) //
+		|| opt_max_operation_size < 0 //
+		|| opt_max_connections < 0 //
+		|| opt_stripe_size < 0)
```

Comment:
Any reason for the formatting changes? Why are lines terminated with `//`?

Reply:
With each backend type, the if-condition gets longer and longer. To improve readability, I put the conditions on separate lines each. The terminating // is there to prevent the automated code formatting tool to merge the lines into a single one.